packages/flutter/lib/src/rendering/platform_view.dart

```diff
@@ -483,6 +483,237 @@ class _UiKitViewGestureRecognizer extends OneSequenceGestureRecognizer {
+  /// any newly arriving events there's nothing we need to invalidate.
+  PlatformViewHitTestBehavior hitTestBehavior;
+
+  /// {@macro flutter.rendering.RenderAndroidView.updateGestureRecognizers}
+  void updateGestureRecognizers(
+      Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers) {
+    assert(gestureRecognizers != null);
+    assert(
+      _factoriesTypeSet(gestureRecognizers).length == gestureRecognizers.length,
+      'There were multiple gesture recognizer factories for the same type, there must only be a single '
+      'gesture recognizer factory for each gesture recognizer type.',
+    );
+    if (_factoryTypesSetEquals(
+        gestureRecognizers, _gestureRecognizer?.gestureRecognizerFactories)) {
+      return;
+    }
+    _gestureRecognizer?.dispose();
+    _gestureRecognizer =
+        _LinuxViewGestureRecognizer(viewController, gestureRecognizers);
+  }
+
+  @override
+  bool get sizedByParent => true;
+
+  @override
+  bool get alwaysNeedsCompositing => true;
+
+  @override
+  bool get isRepaintBoundary => true;
+
+  _LinuxViewGestureRecognizer? _gestureRecognizer;
+
+  PointerEvent? _lastPointerDownEvent;
+
+  @override
+  void performResize() {
+    size = constraints.biggest;
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    context.addLayer(PlatformViewLayer(
+      rect: offset & size,
+      viewId: _viewController.id,
+    ));
+  }
+
+  @override
+  bool hitTest(BoxHitTestResult result, {Offset? position}) {
+    if (hitTestBehavior == PlatformViewHitTestBehavior.transparent || !size.contains(position!))
+      return false;
+    result.add(BoxHitTestEntry(this, position));
+    return hitTestBehavior == PlatformViewHitTestBehavior.opaque;
+  }
+
+  @override
+  bool hitTestSelf(Offset position) =>
+      hitTestBehavior != PlatformViewHitTestBehavior.transparent;
+
+  @override
+  void handleEvent(PointerEvent event, HitTestEntry entry) {
+    if (event is! PointerDownEvent) {
+      return;
+    }
+    _gestureRecognizer!.addPointer(event);
+    _lastPointerDownEvent = event.original ?? event;
+  }
+
+  // This is registered as a global PointerRoute while the render object is attached.
+  void _handleGlobalPointerEvent(PointerEvent event) {
+    if (event is! PointerDownEvent) {
+      return;
+    }
+    if (!(Offset.zero & size).contains(globalToLocal(event.position))) {
+      return;
+    }
+    if ((event.original ?? event) != _lastPointerDownEvent) {
+      // The pointer event is in the bounds of this render box, but we didn't get it in handleEvent.
+      // This means that the pointer event was absorbed by a different render object.
+      // Since on the platform side the FlutterTouchIntercepting view is seeing all events that are
+      // within its bounds we need to tell it to reject the current touch sequence.
+      _viewController.rejectGesture();
+    }
+    _lastPointerDownEvent = null;
+  }
```

Comment:
There appears to be a lot of exactly-duplicated code between the existing two, and now these three, sets of classes. Rather than make a third copy, we should extract new superclasses to put all the common pieces in, and have all of the individual platform implementations inherit from that.

Reply:
I think these may be done in future PR, not in this PR.

Reply:
I'm inclined to agree with @stuartmorgan: two copies is a temporary workaround, three becomes a big buildup of technical debt. I'd rather there were some refactoring here.